### PR TITLE
Fix agent disconnection during upgrades causing the script to hang indefinitely

### PR DESCRIPTION
**Pull request by GGP1**
|Related issue|
|---|
| Closes https://github.com/wazuh/wazuh/issues/20901 |

## Description

Adds the logic necessary to remove failed items from the list and stop waiting for their tasks.

## Logs/Alerts example

<details><summary>Agent disconnection during upgrade</summary>

```console
root@wazuh-master:/# /var/ossec/framework/python/bin/python3 /var/ossec/framework/scripts/agent_upgrade.py -a 001 -f /wpk_file

Upgrading...
...
# Debug logs to demonstrate the behavior, won't be present in production
Task results: AffectedItemsWazuhResult({'dikt': {}, '_affected_items': [{'message': 'Success', 'agent': '001', 'task_id': 1, 'node': 'master-node', 'module': 'upgrade_module', 'command': 'upgrade_custom', 'status': 'Updating', 'create_time': '2024-06-25T14:53:30Z', 'update_time': '2024-06-25T14:53:30Z'}], '_failed_items': {}, '_total_affected_items': 1, '_total_failed_items': 0, '_sort_fields': None, '_sort_casting': ['int'], '_sort_ascending': [True], '_all_msg': 'All upgrade tasks were returned', '_some_msg': 'Some upgrade tasks were not returned', '_none_msg': 'No upgrade task was returned'})
Task results: AffectedItemsWazuhResult({'dikt': {}, '_affected_items': [], '_failed_items': {{'type': 'about:blank', 'title': 'Bad Request', 'code': 1707, 'extra_message': None, 'extra_remediation': None, 'cmd_error': False, 'dapi_errors': {}, 'ids': []}: {'001'}}, '_total_affected_items': 0, '_total_failed_items': 1, '_sort_fields': None, '_sort_casting': ['int'], '_sort_ascending': [True], '_all_msg': 'All upgrade tasks were returned', '_some_msg': 'Some upgrade tasks were not returned', '_none_msg': 'No upgrade task was returned'})

Failed upgrades:
  Agent 001 status: Agent disconnected during the upgrade
```

</details>

<details><summary>Multiple agents disconnection during upgrade</summary>

```console
root@wazuh-master:/# /var/ossec/framework/python/bin/python3 /var/ossec/framework/scripts/agent_upgrade.py -a 001 002 -f /wpk_file

Upgrading...
AffectedItemsWazuhResult({'dikt': {}, '_affected_items': [{'message': 'Success', 'agent': '001', 'task_id': 1, 'node': 'master-node', 'module': 'upgrade_module', 'command': 'upgrade_custom', 'status': 'Updating', 'create_time': '2024-06-25T17:28:55Z', 'update_time': '2024-06-25T17:28:55Z'}, {'message': 'Success', 'agent': '002', 'task_id': 1, 'node': 'master-node', 'module': 'upgrade_module', 'command': 'upgrade_custom', 'status': 'Updating', 'create_time': '2024-06-25T17:28:55Z', 'update_time': '2024-06-25T17:28:55Z'}], '_failed_items': {}, '_total_affected_items': 2, '_total_failed_items': 0, '_sort_fields': None, '_sort_casting': ['int'], '_sort_ascending': [True], '_all_msg': 'All upgrade tasks were returned', '_some_msg': 'Some upgrade tasks were not returned', '_none_msg': 'No upgrade task was returned'})
Task results: AffectedItemsWazuhResult({'dikt': {}, '_affected_items': [], '_failed_items': {{'type': 'about:blank', 'title': 'Bad Request', 'code': 1707, 'extra_message': None, 'extra_remediation': None, 'cmd_error': False, 'dapi_errors': {}, 'ids': []}: {'001', '002'}}, '_total_affected_items': 0, '_total_failed_items': 1, '_sort_fields': None, '_sort_casting': ['int'], '_sort_ascending': [True], '_all_msg': 'All upgrade tasks were returned', '_some_msg': 'Some upgrade tasks were not returned', '_none_msg': 'No upgrade task was returned'})

Failed upgrades:
  Agent 001 status: Agent disconnected during the upgrade
  Agent 002 status: Agent disconnected during the upgrade
```

</details>

### CODE REVIEW

**Review by RamosFe — APPROVED** [review]
LGTM 🚀

**Review by Selutario — CHANGES_REQUESTED** [review]
Take a look at the unit tests, probably one of them should be updated.

File: framework/scripts/agent_upgrade.py [hunk 1]
```diff
@@ -189,6 +190,12 @@ async def check_status(affected_agents: list, result_dict: dict, failed_agents:
                     else task_result['status']
                 result_dict.pop(task_result['agent'])
                 affected_agents.discard(task_result['agent'])
+            
+        for agent_ids in task_results.failed_items.values():
+            if task_result['agent'] in agent_ids:
```
Comment (Selutario) [inline framework/scripts/agent_upgrade.py:195]: I forgot to add a comment on this. Could `task_result` be referenced before assignment? Also, and just to make sure: I guess task_result is using the value of the last item in the previous `for` loop, would this work as expected when upgrading two or more agents?

**Review by Selutario — CHANGES_REQUESTED** [review]
Take a look at the unit tests, probably one of them should be updated.

File: framework/scripts/agent_upgrade.py (hunk 1 above)
Comment (GGP1) [reply framework/scripts/agent_upgrade.py:195]: It was a mistake caused by the fact that I initially had this loop inside the other one, so we were accessing a different agent ID on each iteration. But considering the affected items list may be empty, I took it out.

I changed this so we get the IDs from the `affected_agents` set instead, good catch!

**Review by Selutario — APPROVED** [review]
LGTM

**Pull request merged**